src/components/compound/MenuTab/index.tsx

```diff
@@ -0,0 +1,56 @@
+import type { ReactElement } from 'react';
+import { useState, useEffect, cloneElement, Children } from 'react';
+import { StyledContainer, StyledTab, StyledTabWrapper } from './styled';
+import type { MenuTabProps } from './types';
+
+const MenuTab = ({
+  children,
+  initialOnChild,
+  tabText
+}: MenuTabProps): ReactElement => {
+  const [onContent, setOnContent] = useState(initialOnChild);
+  const [childrenEl, setChildrenEl] = useState<React.ReactFragment[]>();
+
+  const handleContent = (e: React.MouseEvent<HTMLDivElement>): void => {
+    const { id } = e.currentTarget.dataset;
+
+    if (id) {
+      if (onContent === id) return;
+
+      setOnContent(id);
+    }
+  };
```

Comment:
```suggestion
  const handleContent = (e: React.MouseEvent<HTMLDivElement>): void => {
    const { id } = e.currentTarget.dataset;

    if (id) {
      if (onContent === id) return;

      setOnContent(id);
      onChangeContent?.(id);
    }
  };
```
와 같이 content가 바뀔때 상위로 올려보내기 위한 메소드를 prop으로 받는것 어떨까요?
id를 실어 올리게되면, page 단에서 해당 id로 queryString를 바꾼다던가 하는 처리가 가능할것 같습니다.

Reply:
@AlangGY 안 그래도 지금 이부분 좀 다시보고 있었는데...

우선 말씀해주신 onChange는 넘 좋아요 바로 반영하겠씁니다.

근데 if문 모양이 너무 이상해서 고민중인데...
이게 id가 undefined일 수도 있어서 setOnContent(id)가 에러가 뜨더라구요
id가 존재하는 게 확정일 때만 동작할 수 있는 상황입니다.
그래서
```typescript
if (onContent === id) return;

if (id) setOnContent(id);
```

이런 식으로 바꾸는 게 나을지 고민중입니다.

추가로 onContent라는 변수명이 너무 함수 같아서
displayingChildIdx 로 수정할 계획입니다.

Reply:
저희 린트룰 때문에 
```typescript
id && setOnContent(id)
```
는 안되더라구요... 그 제가 설정한 표현식 관련 린트룰....ㅅ..

Reply:
위에 무시해주세요 코드 수정해서 다시 커밋했슴다